src/databricks/sql/thrift_backend.py

```diff
@@ -279,24 +283,40 @@ def attempt_request(attempt):
             # - non-None method_return -> success, return and be done
             # - non-None retry_delay -> sleep delay before retry
             # - error, error_message always set when available
+
+            error = None
             try:
                 logger.debug("Sending request: {}".format(request))
                 response = method(request)
                 logger.debug("Received response: {}".format(response))
                 return response
-            except Exception as error:
+            except socket.timeout as err:
+                # socket.timeout means the connection was timed out by the socket package
+                retry_delay = self._retry_delay_default
+                error_message = str(err)
+                error = err
+            except OSError as err:
+                # OSError 110 means the connection was timed out by the operating system
+                if "Errno 110" in str(err):
+                    retry_delay = self._retry_delay_default
+                else:
+                    retry_delay = None
```

Comment:
Can you initialize this at the beginning and can only be overridden on selected (retriable) errors?

Reply:
Sure. c76ee65ee674a3c1fcd1b45417b0493bcccdf899